logsmanagement/flb_plugin.c

```diff
@@ -0,0 +1,982 @@
+    m_assert(buff->in->data, "buff->in->text cannot be NULL");
+    m_assert(*buff->in->data, "*buff->in->text cannot be 0");
+    m_assert(buff->in->text_size, "buff->in->text_size cannot be 0");
+
+    /* Replace last '\n' with '\0' to null-terminate text */
+    buff->in->data[buff->in->text_size - 1] = '\0'; 
+
+    /* Store status (timestamp and text_size must have already been 
+     * stored during flb_write_to_buff_cb() ). 
+     * IMPORTANT! Systemd logs and docker events logs have already been parsed, 
+     * so in that case set status to DONE. */
+    buff->in->status = (p_file_info->log_type == FLB_SYSTEMD || p_file_info->log_type == FLB_DOCKER_EV) ? 
+                        CIRC_BUFF_ITEM_STATUS_DONE : CIRC_BUFF_ITEM_STATUS_UNPROCESSED;
+
+    /* Load max size of compressed buffer, as calculated previously */
+    size_t text_compressed_buff_max_size = buff->in->text_compressed_size;
+
+    /* Do compression */
+    buff->in->text_compressed = buff->in->data + buff->in->text_size;
+    buff->in->text_compressed_size = LZ4_compress_fast( buff->in->data, buff->in->text_compressed, 
+                                                            buff->in->text_size, text_compressed_buff_max_size, 
+                                                            p_file_info->compression_accel);
+    m_assert(buff->in->text_compressed_size != 0, "Text_compressed_size should be != 0");
+
+    // TODO: Validate compression option?
+
+    circ_buff_insert(buff);
+
+    /* Instruct log parsing and metrics extraction, asynchronously for web logs
+     * but synchronously for systemd logs (also because the later must happen
+     * while p_file_info->flb_tmp_buff_mut is still acquired - not the case for
+     * the web logs) */
+    if(p_file_info->log_type == FLB_GENERIC || p_file_info->log_type == FLB_WEB_LOG){
+        uv_mutex_lock(&p_file_info->notify_parser_thread_mut);
+        p_file_info->log_batches_to_be_parsed++;
+        uv_cond_signal(&p_file_info->notify_parser_thread_cond);
+        uv_mutex_unlock(&p_file_info->notify_parser_thread_mut);
+    } else if(p_file_info->log_type == FLB_SYSTEMD) {
+        uv_mutex_lock(p_file_info->parser_metrics_mut);
+        p_file_info->parser_metrics->num_lines_total += p_file_info->flb_tmp_systemd_metrics.num_lines;
+        p_file_info->parser_metrics->num_lines_rate = p_file_info->flb_tmp_systemd_metrics.num_lines;
+        p_file_info->flb_tmp_systemd_metrics.num_lines = 0;
+        for(int i = 0; i < 9; i++){
+            p_file_info->parser_metrics->systemd->sever[i] = p_file_info->flb_tmp_systemd_metrics.sever[i];
+            p_file_info->flb_tmp_systemd_metrics.sever[i] = 0;
+        }
+        for(int i = 0; i < 25; i++){
+            p_file_info->parser_metrics->systemd->facil[i] = p_file_info->flb_tmp_systemd_metrics.facil[i];
+            p_file_info->flb_tmp_systemd_metrics.facil[i] = 0;
+        }
+        for(int i = 0; i < 193; i++){
```

Comment:
Here in this file we have the same magic numbers, if you create `defines` for the previous one, please, do not forget to adjust this code too.

Reply:
Done.